File: backend/core-service/internal/service/item_service.go
package service

import (
	"log"

	"core-service/internal/dto"
	"core-service/internal/repository"
)

type ItemService struct {
	repo *repository.ItemRepository
}

func NewItemService(repo *repository.ItemRepository) *ItemService {
	return &ItemService{repo: repo}
}

func (s *ItemService) GetAllItems() []dto.Item {
	return s.repo.GetAll()
}

func (s *ItemService) CreateItem(req dto.CreateItemRequest) dto.Item {
	log.Printf("Creating item: %s", req.Name)
	return s.repo.Create(req.Name, req.Description)
}

func (s *ItemService) GetItem(id string) (dto.Item, bool) {
	return s.repo.GetByID(id)
}

func (s *ItemService) DeleteItem(id string) bool {
	return s.repo.Delete(id)
}

func (s *ItemService) UpdateItem(id string, req dto.CreateItemRequest) (dto.Item, bool) {
	return s.repo.Update(id, req.Name, req.Description)
}

File: backend/core-service/src/core_service/service/item_service.py
from core_service.domain import InvalidInputError, Item, ItemStore


class ItemService:
    def __init__(self, store: ItemStore) -> None:
        self._store = store

    def get_all(self) -> list[Item]:
        return self._store.get_all()

    def create(self, name: str, description: str) -> Item:
        if not name or not name.strip():
            raise InvalidInputError("name is required")
        return self._store.create(name, description)

    def get(self, item_id: str) -> Item:
        return self._store.get_by_id(item_id)

    def delete(self, item_id: str) -> None:
        self._store.delete(item_id)

    def ping(self) -> None:
        self._store.ping()
